release-notes/Cabal-3.16.0.0.md

```diff
@@ -0,0 +1,222 @@
+  and the connect of the `.gc` extension to this build-tool.
+
+- Remove knowledge about hmake and haskell-suite [#10912](https://github.com/haskell/cabal/pull/10912)
+
+  The `hmake` tool has long been abandoned, and the `haskell-suite` compiler did not emerge.
+  Knowledge about these tools has been removed from `Cabal`.
+
+- Haddock documentation added to Paths_pkgname and PackageInfo_pkgname modules [#10941](https://github.com/haskell/cabal/pull/10941)
+
+  The automatically-generated modules `Paths_pkgname` and `PackageInfo_pkgname`
+  now include Haddock documentation.
+
+- ghc-options added to js, cmm and asm [#10949](https://github.com/haskell/cabal/pull/10949)
+
+  Now we have the ability to pass ghc-options flags to all component.
+
+  ```
+    js-sources: jsbits/lib.js
+    ghc-options: -optJSP-your_flag
+  ```
+
+- Adding JavaScript Preprocessor Support [#10967](https://github.com/haskell/cabal/pull/10967)
+
+  Adding new syntax jspp-options that would support the -optJSP preprocessor flag for js sources.
+
+  ```
+    js-sources: jsbits/lib.js
+    jspp-options: -DYOUR_FLAG
+  ```
+
+- Fix parsing the AbiTag with development versions of GHC [#10170](https://github.com/haskell/cabal/issues/10170) [#10979](https://github.com/haskell/cabal/pull/10979)
+
+  - Allow parsing the AbiTag for developmement versions of GHC.
+
+- Take `--working-dir` into account in runPreProcessorWithHsBootHack [#11000](https://github.com/haskell/cabal/issues/11000) [#10991](https://github.com/haskell/cabal/pull/10991)
+
+  The preprocessor hs-boot hack handles the situation in which a file to be
+  preprocessed is supplied alongside an hs-boot file, e.g. Foo.x and Foo.hs-boot
+  are given together.
+  
+  This code now respects the --working-dir Cabal global argument. This fixes
+  build failures of the form:
+  
+    attempting to use module `Foo` (Foo.hs-boot) which is not loaded
+
+- Haddock: don't try to copy build dir if it doesn't exist [#11001](https://github.com/haskell/cabal/issues/11001) [#10992](https://github.com/haskell/cabal/pull/10992)
+
+  This small patch fixes a little oversight in 'reusingGHCCompilationArtifacts',
+  which would unconditionally attempt to copy over the GHC build artifacts to be
+  re-used by Haddock, even when those artifacts did not exist (which caused
+  an error).
+
+- add SPDX license data version 3.26 [#11014](https://github.com/haskell/cabal/pull/11014)
+
+  - Update SPDX license list to version 3.26.0 2024-12-30
+
+- Use response files when calling `ghc` [#9367](https://github.com/haskell/cabal/pull/9367)
+
+  Cabal now passes GHC arguments through a response file. This prevents an error
+  where very large packages would fail to build due to the module names exceeding
+  the `ARG_MAX` command-line length limit:
+
+  ```
+  ghc: createProcess: posix_spawnp: resource exhausted (Argument list too long)
+  ```
+
+  Notes:
+  - Response files will not be used if the first argument to `ghc` is
+    `--interactive`.
```

Comment:
Should we note here that the behavior was changed by a subsequent PR (above, since this list is in reverse order; see `--with-repl`)?

Reply:
I am just axing this bit, not to confuse the user.